### PR TITLE
Feature/cronapp-3720-develop-Utilização de Cookies 

### PR DESCRIPTION
**Pull request by alealcronapp**
#Adicionando parametro para unidade de medida de tempo (Segundos, horas, minutos, dias).

Feature ou Hotfix/CRONAPP-3720 - Utilização de Cookies

https://cronapp.atlassian.net/browse/CRONAPP-3720

## Problema:
CRONAPP-3720 - Utilização de Cookies
## Melhoria:
Utilização de Cookies no lado cliente e servidor para salvar e recuperar mensagem ( em ambos os lados)
## Solução:
Criação das funções para setar e obter cookie nas camadas cliente e servidor.

branches feature/CRONAPP-3720-Develop, origin/feature/CRONAPP-3720-Develop

### CODE REVIEW

File: cronapi.js [hunk 1]
```diff
@@ -6049,6 +6049,55 @@ function cronapi() {
...
+   * @param {ObjectType.STRING} measurementUnit {{measurementUnit}} 
+   * @param {ObjectType.LONG} expires {{expirationTime}}
+   */
+   this.cronapi.util.setCookie = function (key,value,/**@type {ObjectType.STRING} @description {{measurementUnit}} @blockType util_dropdown @keys seconds|minutes|hours|days @values {{seconds}}|{{minutes}}|{{hours}}|{{days}}  */ measurementUnit ,expires) {
+    
+    let now = new Date(); 
+    let expiration = new Date();
+    let factor = 1;
+
+    if (measurementUnit === 'seconds') {
```
Comment (drigores) [inline cronapi.js:6069]: Need be refactored into a method who maybe could be called as: 'getFactoryByMeasurementUnit'

Clean Code Readability Principle
Clean Code Meaningful Names principle
Clean Code Function principle

File: cronapi.js [hunk 2]
```diff
@@ -6049,6 +6049,55 @@ function cronapi() {
...
+    } else if(measurementUnit ==='minutes') {
+      factor = 60000;
+    } else if(measurementUnit ==='hours') {
+      factor = 3600000;
+    } else if(measurementUnit ==='days') {
+      factor = 86400000;
+    }
+
+    if (this.cronapi.logic.isNullOrEmpty(expires) || expires == 0) {      
+      expiration.setTime(now.getTime() + 1000*60*60*24);
```
Comment (drigores) [inline cronapi.js:6080]: What this 1000*60*60*24 means? give it a variable with const and meaningful name.
const DAY_IN_MILLISECONDS for example.
Code must be readable.

File: cronapi.js [hunk 3]
```diff
@@ -6049,6 +6049,55 @@ function cronapi() {
...
+    } else if(measurementUnit ==='days') {
+      factor = 86400000;
+    }
+
+    if (this.cronapi.logic.isNullOrEmpty(expires) || expires == 0) {      
+      expiration.setTime(now.getTime() + 1000*60*60*24);      
+    } else {    
+      expiration.setTime(now.getTime() + parseInt(expires * factor));
+    }
+      document.cookie = encodeURIComponent(key)+"="+encodeURIComponent(value)+ "; expires="+expiration.toGMTString();
```
Comment (drigores) [inline cronapi.js:6084]: And the OTHERS cookies that exists ?
Need be refactored for something readable for humans.
What this code are doing?
Clean Code Readability Principle
Clean Code Meaningful Names principle
Clean Code Function principle

File: cronapi.js [hunk 4]
```diff
@@ -6049,6 +6049,55 @@ function cronapi() {
...
+     * @type function
+     * @name {{getGookie}}
+     * @nameTags storage | cookie | armazenamento
+     * @description {{getCookieDesc}}
+     * @param {ObjectType.STRING} key {{key}}
+     * @returns {ObjectType.STRING}
+     */
+     this.cronapi.util.getCookie = function(key) {  
+      const value = document.cookie; 
+      const parts = value.split(encodeURIComponent(key)+'=');
```
Comment (drigores) [inline cronapi.js:6097]: WTF this code are doing? 
please reafctor it into readable code.

File: cronapi.js [hunk 5]
```diff
@@ -6049,6 +6049,55 @@ function cronapi() {
...
+     * @nameTags storage | cookie | armazenamento
+     * @description {{getCookieDesc}}
+     * @param {ObjectType.STRING} key {{key}}
+     * @returns {ObjectType.STRING}
+     */
+     this.cronapi.util.getCookie = function(key) {  
+      const value = document.cookie; 
+      const parts = value.split(encodeURIComponent(key)+'='); 
+      if (parts.length === 2) 
+      return decodeURIComponent(parts.pop().split(';').shift());
```
Comment (drigores) [inline cronapi.js:6099]: same for here.
And plus, is this the best logic for it?

**Review by drigores — CHANGES_REQUESTED** [review]

File: cronapi.js (hunk 3 above)
Comment (alealcronapp) [reply cronapi.js:6084]: Discutido em call.

File: cronapi.js (hunk 1 above)
Comment (alealcronapp) [reply cronapi.js:6069]: Feito!

File: cronapi.js (hunk 2 above)
Comment (alealcronapp) [reply cronapi.js:6080]: Feito!

**Pull request merged**